### PR TITLE
Add new installer types

### PR DESCRIPTION
**Pull request by erri120**
Part of #1763, this adds our new "installer" types to go from library items to loadout items.

### CODE REVIEW

File: src/Abstractions/NexusMods.Abstractions.Installers/LoadoutDetails.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{
+    /// <summary>
+    /// Gets the loadout.
+    /// </summary>
+    public required Loadout.ReadOnly Loadout { get; init; }
+
+    /// <summary>
+    /// Gets the game installation.
+    /// </summary>
+    public GameInstallation GameInstallation => Loadout.InstallationInstance;
```
Comment (Al12rs) [inline src/Abstractions/NexusMods.Abstractions.Installers/LoadoutDetails.cs:21]: Is there a reason to pack this data together apart from passing one less parameter?

Comment (erri120) [reply src/Abstractions/NexusMods.Abstractions.Installers/LoadoutDetails.cs:21]: Our current installers have `ModInstallerInfo` which contains a lot more properties. The new installer types don't need most of what's in there. I don't know yet what installers want in addition to this information, so I just made this new record for loadout related details.

Comment (halgari) [reply src/Abstractions/NexusMods.Abstractions.Installers/LoadoutDetails.cs:21]: The installation is in the Loadout: `loadout.InstallationInstance`

Comment (erri120) [reply src/Abstractions/NexusMods.Abstractions.Installers/LoadoutDetails.cs:21]: I can probably just pass `Loadout.ReadOnly` for now.

File: src/Abstractions/NexusMods.Abstractions.Installers/ALibraryArchiveInstaller.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        return ExecuteAsync(libraryArchive, transaction, loadoutDetails, cancellationToken);
+    }
+
+    /// <inheritdoc/>
+    public abstract ValueTask<LoadoutItem.New[]> ExecuteAsync(
+        LibraryArchive.ReadOnly libraryArchive,
+        ITransaction transaction,
+        LoadoutDetails loadoutDetails,
+        CancellationToken cancellationToken);
+}
```
[the remark this answers is not in the document]
Comment (erri120) [reply src/Abstractions/NexusMods.Abstractions.Installers/ALibraryArchiveInstaller.cs:55]: You'd get the tree from the `libraryArchive`. It's on the TODO list in #1763 to have a tree for this.

**Review by Al12rs — APPROVED** [review]

**Pull request merged**